src/main/java/org/reaktivity/nukleus/kafka/internal/memory/DefaultMemoryManager.java

```diff
@@ -32,39 +32,56 @@
     private final BTreeFW btreeRO;
 
     private final int blockSizeShift;
-    private final int maximumBlockSize;
     private final int maximumOrder;
 
-    private final MutableDirectBuffer memoryBuffer;
+    private final MutableDirectBuffer[] memoryBuffers;
+    private final int firstBufferCapacity;
+    private final int addressShift;
+    private final long addressMask;
     private final AtomicBuffer metadataBuffer;
 
 
     public DefaultMemoryManager(
         MemoryLayout memoryLayout)
     {
-        final int minimumBlockSize = memoryLayout.minimumBlockSize();
-        final int maximumBlockSize = memoryLayout.maximumBlockSize();
+        final long minimumBlockSize = memoryLayout.minimumBlockSize();
+        final long maximumBlockSize = memoryLayout.maximumBlockSize();
 
-        this.memoryBuffer = memoryLayout.memoryBuffer();
+        this.memoryBuffers = memoryLayout.memoryBuffers();
         this.metadataBuffer = memoryLayout.metadataBuffer();
         this.blockSizeShift = numberOfTrailingZeros(minimumBlockSize);
-        this.maximumBlockSize = maximumBlockSize;
         this.maximumOrder = numberOfTrailingZeros(maximumBlockSize) - numberOfTrailingZeros(minimumBlockSize);
         this.btreeRO = new BTreeFW(maximumOrder).wrap(metadataBuffer, BTREE_OFFSET, metadataBuffer.capacity() - BTREE_OFFSET);
+        firstBufferCapacity = memoryBuffers[0].capacity();
+        addressShift = Integer.numberOfTrailingZeros(firstBufferCapacity);
+        addressMask = firstBufferCapacity - 1;
     }
 
     @Override
     public long resolve(
         long address)
     {
+        MutableDirectBuffer memoryBuffer;
+
+        if (address < firstBufferCapacity)
+        {
+            memoryBuffer = memoryBuffers[0];
+        }
+        else
+        {
+            int index = (int) (address >> addressShift);
+            memoryBuffer = memoryBuffers[index];
+            address &= addressMask;
+        }
+
         return memoryBuffer.addressOffset() + address;
     }
 
     @Override
     public long acquire(
-        int capacity)
+        final int capacity)
     {
-        if (capacity > this.maximumBlockSize)
+        if (capacity > this.firstBufferCapacity)
```

Comment:
Since `maximumBlockSize` is already guarded by implementation detail of `memoryBuffer.capacity`, shouldn't this remain as `if (capacity > this.maximumBlockSize) ...` ?

Reply:
The meaning of maximumBlockSize in MemoryLayout is unchanged: it is the overall size of the cache.  I  decided not to try to make a distinction between capacity and maximumBlockSize,  because the information is already conveyed in the array of MemoryBuffers returned by MemoryLayout.getMemoryBuffers().

I could rename `maximumBlockSize` to `capacity` in MemoryLayout  if you think that would clarify matters.  The only downside I can see is it might then be confused with the capacity parameter to MemoryManager.acquire.